api/proto/config/cipherinfo.proto

```diff
@@ -0,0 +1,43 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+option go_package  = "github.com/lf-edge/eve/api/go/config";
+option java_package = "com.zededa.cloud.uservice.proto";
+
+// Cipher information for sensitive data.
+
+// Security Key Exchange Method
+enum KeyExchangeScheme {
+    KEA_NONE = 0;
+    KEA_ECDH = 1;
+}
+
+// Encryption Scheme for Cipher Payload
+enum EncryptionScheme {
+    SA_NONE = 0;
+    SA_AES_256_CFB = 1;
+}
+
+message CipherInfo {
+    string id = 100;
+
+    KeyExchangeScheme keyExchangeScheme = 1;
+    EncryptionScheme encryptionScheme = 2;
+
+    // Initial Value for ECDH computation
+    bytes initialValue = 3;
+
+    // controller certificate
+    bytes controllerCert = 4;
+
+    // controller certificate sha
+    string controllerCertSha256 = 5;
+
+    // device certificate sha
+    string deviceCertSha256 = 6;
```

Comment:
Based on the discussion today I think we can capture this using a 'int CertContextID' which is used to refer to the cert(s).

Reply:
Actually we are now using bytes deviceCertHash plus an enum (in auth.proto) to identify the certificate(s) in the envelop. Makes sense to do the same here.
string doesn't work for byte strings - protobuf requires them to be ASCII/UTF-8.